batched/dense/unit_test/Test_Batched_SerialLaswp.hpp

```diff
@@ -0,0 +1,509 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+/// \author Yuuichi Asahi (yuuichi.asahi@cea.fr)
+#include <gtest/gtest.h>
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+
+#include "KokkosBatched_Util.hpp"
+#include "KokkosBatched_Laswp.hpp"
+#include "Test_Batched_DenseUtils.hpp"
+
+using namespace KokkosBatched;
+
+namespace Test {
+namespace Laswp {
+
+template <typename DeviceType, typename PivViewType, typename AViewType, typename ArgDirect>
+struct Functor_BatchedSerialLaswp {
+  using execution_space = typename DeviceType::execution_space;
+  PivViewType m_ipiv;
+  AViewType m_a;
+
+  KOKKOS_INLINE_FUNCTION
+  Functor_BatchedSerialLaswp(const PivViewType &ipiv, const AViewType &a) : m_ipiv(ipiv), m_a(a) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const ArgDirect &, const int k, int &info) const {
+    auto sub_ipiv = Kokkos::subview(m_ipiv, k, Kokkos::ALL);
+    if constexpr (AViewType::rank == 3) {
+      auto sub_a = Kokkos::subview(m_a, k, Kokkos::ALL, Kokkos::ALL);
+      info += KokkosBatched::SerialLaswp<ArgDirect>::invoke(sub_ipiv, sub_a);
+    } else {
+      auto sub_a = Kokkos::subview(m_a, k, Kokkos::ALL);
+      info += KokkosBatched::SerialLaswp<ArgDirect>::invoke(sub_ipiv, sub_a);
+    }
+  }
+
+  inline int run() {
+    using value_type = typename AViewType::non_const_value_type;
+    std::string name_region("KokkosBatched::Test::SerialLaswp");
+    const std::string name_value_type = Test::value_type_name<value_type>();
+    std::string name                  = name_region + name_value_type;
+    int info_sum                      = 0;
+    Kokkos::Profiling::pushRegion(name.c_str());
```

Comment:
This profiling region can be removed (matches the inner `parallel_reduce` exactly)